### DISCUSSION TITLE
Making Electron builds 40x smaller

### DISCUSSION THREAD

**Discussion by portasynthinca3** [benawad/dogehouse#1252] [Show and tell]
I've recently been working on a project of mine called [hrt](https://github.com/portasynthinca3/hrt). Well, "working" is a strong word as I've spent at most 12 hours on it in total, but in my opinion it's enough to be considered a proof of concept.

Just like Java applications never carry the heavy JVM and supporting libraries required to run them, the main idea here is to avoid including Electron binaries in the distributions.

Key advantages of this approach:
  - **A lot of bandwidth and disk space is saved.** Not everyone has a speedy Internet connection to sit around a wait for a 50-100MB binary to download per each app that uses Electron.
  - **Portable, OS- and architecture-agnostic.** One universal file for every platform out there, as long as there is an Electron runtime for it. Your average Joe knows what OS they're running, but with the introduction of Apple Silicon in Macs this might introduce some confusion for everyday users.
  - **Blazing fast build times** because you're only producing a single artifact that's just a glorified tar-gzipped source tree.

Key disadvantages of this approach:
  - **A runtime that's not included with the distribution is required** to run DogeHouse.
  - **It only makes sense if others are using it too.** Otherwise DogeHouse users might go through an unnecessary two-step installation process.
  - **It's literally something I created on a weekend**, nobody has used it so far, it's immature.

**Comment by thestonechat** [comment]
This would be cool if it didn't need to download extra stuff and made installation "difficult"...
Cool concept tho, if there's a workaround with that I'd honestly love on work on that too ^^
Maybe keep working on it 😄

**Reply by portasynthinca3** [reply]
Unfortunately, I don't really see a way get rid of the Electron runtime completely, be it integrated into the app itself (as it is common now) or exported into a separate package (as I'm proposing) except for integration of Electron into all three major desktop operating systems' kernels.
